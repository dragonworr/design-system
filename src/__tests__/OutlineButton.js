import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { OutlineButton, theme } from '..'

describe('OutlineButton', () => {
  test('renders', () => {
    const json = renderer.create(<OutlineButton theme={theme} />).toJSON()
    expect(json).toMatchSnapshot()
  })
})
