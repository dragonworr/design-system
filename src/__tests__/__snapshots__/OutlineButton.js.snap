// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutlineButton disabled prop sets 1`] = `
.c1 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  text-decoration: none;
  font-family: inherit;
  font-weight: 600;
  cursor: pointer;
  border-radius: 2px;
  background-color: #007aff;
  color: #fff;
  border-width: 0;
  border-style: solid;
  height: 40px;
  font-size: 14px;
  padding: 0 18px;
}

.c1:disabled {
  opacity: 0.25;
}

.c0 {
  background-color: transparent;
  border-width: 2px;
  border-style: solid;
  color: #007aff;
  border-color: #007aff;
}

.c0:hover {
  background-color: transparent;
}

<button
  className="c0 c1"
  disabled={true}
/>
`;

exports[`OutlineButton renders 1`] = `
.c1 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  text-decoration: none;
  font-family: inherit;
  font-weight: 600;
  cursor: pointer;
  border-radius: 2px;
  background-color: #007aff;
  color: #fff;
  border-width: 0;
  border-style: solid;
  height: 40px;
  font-size: 14px;
  padding: 0 18px;
}

.c1:disabled {
  opacity: 0.25;
}

.c1:hover {
  background-color: #049;
}

.c0 {
  background-color: transparent;
  border-width: 2px;
  border-style: solid;
  color: #007aff;
  border-color: #007aff;
}

.c0:hover {
  color: #049;
  border-color: #049;
  background-color: transparent;
}

<button
  className="c0 c1"
/>
`;

exports[`OutlineButton without disabled prop sets 1`] = `
.c1 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  text-decoration: none;
  font-family: inherit;
  font-weight: 600;
  cursor: pointer;
  border-radius: 2px;
  background-color: #007aff;
  color: #fff;
  border-width: 0;
  border-style: solid;
  height: 40px;
  font-size: 14px;
  padding: 0 18px;
}

.c1:disabled {
  opacity: 0.25;
}

.c1:hover {
  background-color: #049;
}

.c0 {
  background-color: transparent;
  border-width: 2px;
  border-style: solid;
  color: #007aff;
  border-color: #007aff;
}

.c0:hover {
  color: #049;
  border-color: #049;
  background-color: transparent;
}

<button
  className="c0 c1"
/>
`;
